src/iam_roles/iamrolesservice.py

```diff
@@ -138,13 +162,14 @@ def lambda_handler(event, context):
                 if isinstance(detail, dict):
                     # extract the "Instance_Region" and "Instance" fields
                     instance_region = detail["Instance_Region"]
+                    instance_region_name = region_names.get(instance_region, instance_region)
                     instance = detail["Instance"]
                     ec2 = "ec2:instance/" + instance
                     response = cost_of_instance(
                         event, client, instance, start_date, end_date
                     )
 
-                    ec2_resource = boto3.resource("ec2", region_name=instance_region)
+                    ec2_resource = boto3.resource("ec2", region_name=instance_region_name)
```

Comment:
I doubt it will not work by providing region name here all boto3 library works for region id here region id will be passed

Reply:
And region information need to be updated in line # 132 where region is assigned

Reply:
I think you are seeing old commit, can you please review my latest commit. Thanks.

131ca5a9d982dca11cc2842e2da5fe3dcbf159eb